.github/workflows/my-ci-pipeline.yml

```diff
@@ -55,7 +55,38 @@ jobs:
     - name: Push Docker Prod Image v2
       run: docker push rsalhan/m8-todo-app:prod
 
-    - name: curl Command
-      env:
-        webhook_url: ${{ secrets.WEBHOOK_URL }}
-      run: curl --fail -dH -X POST $webhook_url
+  terraform:
+    name: Terraform Init/Plan/Apply
+    runs-on: ubuntu-latest
+    needs: job-two
+
+    env:
+      # TF_VAR_example: 
+      # TF_CLI_ARGS_apply: "-auto-approve"
+      ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+      ARM_CLIENT_SECRET: ${{ secrets.ARM_CLIENT_SECRET }}
+      ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+      ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+      TF_VAR_OAUTH_CLIENT_ID: ${{ secrets.TF_VAR_OAUTH_CLIENT_ID }}
+      TF_VAR_OAUTH_CLIENT_SECRET: ${{ secrets.TF_VAR_OAUTH_CLIENT_SECRET }}
+      TF_VAR_SECRET_KEY: ${{ secrets.TF_VAR_SECRET_KEY }}
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v4
+
+    # - name: TF setup
+    #   uses: hashicorp/setup-terraform@v1
+    #   with: 
+    #     terraform_version: latest
+
+    - name: TF Init
+      run: terraform init
+      
+    - name: TF Plan
+      run: terraform plan -out=tfplan
```

Comment:
Minor, but:
* If the pipeline is going to automatically `apply` directly after this plan regardless of outcome, is there any point actually running a `plan`? `apply` will complete this step anyway
* If you do want to separate the two steps*, then we typically want to ensure the `apply` actually uses the plan that was generated here! I believe the syntax would then be:
```bash
terraform apply -auto-approve tfplan
```

*For example, if we had some custom checks set up to run against the plans, or if we wanted to archive them for future reference

Reply:
Thanks Jack, for now I've added the plan to the apply line as you've suggested